src/js/angular/sparql-template/controllers.js

```diff
@@ -0,0 +1,360 @@
+                });
+            });
+    };
+
+    // Check if warning message should be shown or removed on repository change
+    const repoIsSetListener = $scope.$on('repositoryIsSet', function () {
+        $scope.setRestricted();
+    });
+
+    window.addEventListener('beforeunload', removeRepoIsSetListener);
+
+    function removeRepoIsSetListener() {
+        repoIsSetListener();
+        window.removeEventListener('beforeunload', removeRepoIsSetListener);
+    }
+}
+
+SparqlTemplateCreateCtrl.$inject = ['$scope', '$location', 'toastr', '$repositories', '$window', '$timeout', 'SparqlTemplatesRestService', 'RDF4JRepositoriesRestService', 'SparqlRestService', 'UriUtils'];
+
+function SparqlTemplateCreateCtrl($scope, $location, toastr, $repositories, $window, $timeout, SparqlTemplatesRestService, RDF4JRepositoriesRestService, SparqlRestService, UriUtils) {
+
+    $scope.templateID = $location.search().templateID || '';
+    $scope.title = ($scope.templateID ? 'Edit' : 'Create') + ' SPARQL Template';
+    $scope.getNamespaces = getNamespaces;
+    $scope.setLoader = setLoader;
+    $scope.addKnownPrefixes = addKnownPrefixes;
+    $scope.noPadding = {paddingRight: 0, paddingLeft: 0};
+    $scope.currentTabConfig = {};
+    // This property is obligatory in order to show YASQUE and YASR properly
+    $scope.orientationViewMode = true;
+    $scope.currentQuery = {};
+
+
+    $scope.$watch(function () {
+        return $repositories.getActiveRepository();
+    }, function (newValue, oldValue) {
+        if (newValue !== oldValue) {
+            $location.path('/sparql-templates');
+        }
+    });
+
+    let timer = null;
+    $scope.goBack = function () {
+        timer = $timeout(function () {
+            $window.history.back();
+        }, 1000);
+    };
+
+    const locationChangeListener = $scope.$on('$locationChangeStart', function (event) {
+        confirmExit(event);
+    });
+
+    window.addEventListener('beforeunload', showBeforeunloadMessage);
+
+    function showBeforeunloadMessage(event) {
+        if (!$scope.currentQuery.isPristine) {
+            event.returnValue = true;
+        }
+    }
+
+    function confirmExit(event) {
+        if (!$scope.currentQuery.isPristine) {
+            if (!confirm('You have unsaved changes. Are you sure that you want to exit?')) {
+                event.preventDefault();
+            } else {
+                window.removeEventListener('beforeunload', showBeforeunloadMessage);
+                locationChangeListener();
+                $timeout.cancel(timer);
+            }
+        }
+    }
+
+    $scope.$on('$destroy', function (event) {
+        window.removeEventListener('beforeunload', showBeforeunloadMessage);
+        locationChangeListener();
+        $timeout.cancel(timer);
+    });
+
+    const defaultTabConfig = {
+        templateID: '',
+        query: 'PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n' +
```

Comment:
The default query should be something more generic like:
`PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX ex: <http://example.com#>
DELETE {
    ?subject ex:myPredicate ?oldValue .
} INSERT {
    ?subject ex:myPredicate ?newValue .
} WHERE {
    ?id rdf:type ex:MyType .
    ?subject ex:isRelatedTo ?id .
}`